packages/hint/src/lib/utils/dom/traverse.ts

```diff
@@ -0,0 +1,36 @@
+import { JSDOMAsyncHTMLElement } from '../../types/jsdom-async-html';
+import { JSDOM } from 'jsdom';
+import { Engine } from '../../engine';
+import { TraverseUp, TraverseDown, Event } from '../../types/events';
+
+const traverseAndNotify = async (element: HTMLElement, dom: JSDOM, engine: Engine, resource: string): Promise<void> => {
+
+    await engine.emitAsync(`element::${element.tagName.toLowerCase()}` as 'element::*', {
+        element: new JSDOMAsyncHTMLElement(element, dom),
+        resource
+    });
+
+    const traverseEvent = {
+        element: new JSDOMAsyncHTMLElement(element, dom),
+        resource
+    } as TraverseDown | TraverseUp;
+
+    await engine.emitAsync(`traverse::down`, traverseEvent);
```

Comment:
Random thought, but should we just get rid of `traverse::down` and `traverse::up`? Seems like a lot of extra events that nothing listens to (with the sole exception of `analyze.ts` which sets a status message and would probably be just fine without them).

I realize this would be breaking, so I can file and track it in a separate issue if everyone agrees (although I'm betting this PR will be breaking already so maybe that's not an issue...)

Reply:
I'd rather not remove them TBH. Us not using them doesn't mean we will not later or that they aren't being used by someone else.

These events could be useful to know if an element is inside another one and similar checks.